Apps/WebClient/src/NewClientApp/src/components/site/FeedbackComponent.vue

```diff
@@ -0,0 +1,272 @@
+<script setup lang="ts">
+import { useVuelidate } from "@vuelidate/core";
+import { maxLength, minLength, required } from "@vuelidate/validators";
+import { computed, ref } from "vue";
+
+import HgButtonComponent from "@/components/common/HgButtonComponent.vue";
+import HgIconButtonComponent from "@/components/common/HgIconButtonComponent.vue";
+import { container } from "@/ioc/container";
+import { SERVICE_IDENTIFIER } from "@/ioc/identifier";
+import { ResultError } from "@/models/errors";
+import User from "@/models/user";
+import { IUserFeedbackService } from "@/services/interfaces";
+import { useErrorStore } from "@/stores/error";
+import { useUserStore } from "@/stores/user";
+
+defineExpose({
+    showDialog,
+});
+
+const userFeedbackService = container.get<IUserFeedbackService>(
+    SERVICE_IDENTIFIER.UserFeedbackService
+);
+
+const userStore = useUserStore();
+const errorStore = useErrorStore();
+
+const comment = ref("");
+const visible = ref(false);
+const hasSubmitted = ref(false);
+const isSuccess = ref(false);
+const isLoading = ref(false);
+
+const validations = computed(() => ({
+    comment: {
+        required,
+        minLength: minLength(5),
+        maxLength: maxLength(500),
+    },
+}));
+
+const v$ = useVuelidate(validations, { comment });
+
+const user = computed<User>(() => userStore.user);
+const hasEmail = computed<boolean>(
+    () => user.value.verifiedEmail && user.value.hasEmail
+);
+const isInvalid = computed<boolean>(
+    () => !v$.value.$dirty || (v$.value.$dirty && v$.value.$invalid)
+);
+const resultTitle = computed<string>(() => {
+    if (hasSubmitted.value) {
+        return isSuccess.value ? "Received" : "Sorry";
+    }
+
+    return "";
+});
+const resultDescription = computed<string>(() => {
+    if (hasSubmitted.value) {
+        return isSuccess.value
+            ? "Your message has been sent successfully!"
+            : "Your message could not be sent out!";
+    }
+
+    return "";
+});
+
+function showDialog(): void {
+    visible.value = true;
+}
+
+function hideDialog(): void {
+    visible.value = false;
+}
+
+function onSubmit(): void {
+    isLoading.value = true;
+    if (isInvalid.value) {
+        v$.value.$touch();
+        isLoading.value = false;
+        return;
+    }
+
+    userFeedbackService
+        .submitFeedback(user.value.hdid, {
+            comment: comment.value,
+        })
+        .then((result) => {
+            isSuccess.value = result;
+        })
+        .catch((err: ResultError) => {
+            if (err.statusCode === 429) {
+                errorStore.setTooManyRequestsError("page");
+            }
+            isSuccess.value = false;
+        })
+        .finally(() => {
+            hasSubmitted.value = true;
+            isLoading.value = false;
+        });
+}
+
+const commentErrors = computed(() =>
+    v$.value.comment.$errors.map((e) =>
+        typeof e.$message === "string" ? e.$message : e.$message.value
+    )
+);
```

Comment:
```suggestion
const commentErrors = computed(() =>
    v$.value.comment.$errors.map((e) => unref(e.$message))
);
```
This can be simplified.  Note: `unref` will have to be improted from "vue".

Reply:
oh nice I didn't know about this 👍